### PR TITLE
ivy.avg_pool1d

### LINKED ISSUE #5174: ivy.avg_pool1d

### PR DESCRIPTION
**Pull request by Change-m**
Close #5174

### CODE REVIEW

**Comment by Change-m** [comment]
avg_pool1d function is belong to Neural Network which is wildly used in Tensorflow, implemented as mixed function that's why it is in extension.py file.

File: ivy/functional/backends/numpy/extensions.py [hunk 1]
```diff
@@ -58,3 +58,19 @@ def native_sparse_array_to_indices_values_and_shape(x):
...
+    ksize:Union[int, float],
+    strides:Union[int, float],
+    padding:int,
+    data_format:str,
+    *,
+    dtype: np.dtype,
+    device: str,
+    out: Optional[np.ndarray] = None
+) ->np.ndarray:
+    return np.avg_pool1d(x, ksize, strides, padding, data_format).astype(x.dtype)
```
Comment (zaeemansari70) [inline ivy/functional/backends/numpy/extensions.py:76]: Same for numpy backend. no function such as `avg_pool1d()`.
This also needs to be implemented compositionally.

File: ivy/functional/backends/jax/extensions.py [hunk 2]
```diff
@@ -56,3 +56,17 @@ def native_sparse_array_to_indices_values_and_shape(x):
...
+    /,
+    ksize:Union[int, float],
+    strides:Union[int, float],
+    padding:int,
+    data_format:str,
+    *,
+    dtype: jnp.dtype,
+    out: Optional[JaxArray] = None
+) ->JaxArray:
+    return jnp.avg_pool1d(x, ksize, strides, padding, data_format, dtype=dtype)
```
Comment (zaeemansari70) [inline ivy/functional/backends/jax/extensions.py:72]: There doesn't seem to be an `avg_pool1d()` in the jax backend.
This needs to be implemented compositionally.

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Your PR looks amazing ✨!
Just requested a few changes and left comments on them which needs to be addressed before we can merge your PR! 🙂
Thanks!

**Comment by zaeemansari70** [comment]
Closing this as discussed 🙂

**Pull request closed without merging**